pkg/operator/controller/controller.go

```diff
@@ -148,72 +151,20 @@ func (r *reconciler) reconcile() (reconcile.Result, error) {
 // ensureRouterNamespace ensures all the necessary scaffolding exists for
 // routers generally, including a namespace and all RBAC setup.
 func (r *reconciler) ensureRouterNamespace() error {
-	cr, err := r.ManifestFactory.RouterClusterRole()
-	if err != nil {
-		return fmt.Errorf("failed to build router cluster role: %v", err)
-	}
-	err = r.Client.Get(context.TODO(), types.NamespacedName{Namespace: cr.Namespace, Name: cr.Name}, cr)
-	if err != nil {
-		if !errors.IsNotFound(err) {
-			return fmt.Errorf("failed to get router cluster role %s: %v", cr.Name, err)
-		}
-		err = r.Client.Create(context.TODO(), cr)
-		if err == nil {
-			logrus.Infof("created router cluster role %s", cr.Name)
-		} else if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create router cluster role %s: %v", cr.Name, err)
-		}
+	if err := r.ensureRouterClusterRole(); err != nil {
```

Comment:
This isn't namespaced, would it make sense to move it out of `ensureRouterNamespace`?